lib/lift_wing_api.rb

```diff
@@ -72,36 +103,103 @@ def lift_wing_server
-                      project_model: @project_quality_model,
+      sentry_extra: { rev_ids:, project_code: wiki_key,
+                      project_model: model_key,
                       error_count: @errors.count })
   end
+
+  def deleted?(response)
+    LiftWingApi::DELETED_REVISION_ERRORS.any? do |revision_error|
+      response.dig('error').include?(revision_error)
+    end
+  end
+
+  # LiftWing articlequality ratings are often derived from the en.wiki system,
+  # so this is the fallback scheme.
+  ENWIKI_WEIGHTING = { 'FA'    => 100,
+                       'GA'    => 80,
+                       'B'     => 60,
+                       'C'     => 40,
+                       'Start' => 20,
+                       'Stub'  => 0 }.freeze
+  FRWIKI_WEIGHTING = { 'adq' => 100,
+                       'ba' => 80,
+                       'a' => 60,
+                       'b' => 40,
+                       'bd' => 20,
+                       'e' => 0 }.freeze
+  TRWIKI_WEIGHTING = { 'sm' => 100,
+                       'km' => 80,
+                       'b' => 60,
+                       'c' => 40,
+                       'baslagıç' => 20,
+                       'taslak' => 0 }.freeze
+  RUWIKI_WEIGHTING = { 'ИС' => 100,
+                       'ДС' => 80,
+                       'ХС' => 80,
+                       'I' => 60,
+                       'II' => 40,
+                       'III' => 20,
+                       'IV' => 0 }.freeze
+  PTWIKI_WEIGHTING = { '6' => 100,
+                       '5' => 80,
+                       '4' => 60,
+                       '3' => 40,
+                       '2' => 20,
+                       '1' => 0 }.freeze
+  UKWIKI_WEIGHTING = { 'ДС' => 100,
+                       'ВС' => 80,
+                       'I' => 60,
+                       'II' => 40,
+                       'III' => 20,
+                       'IV' => 0 }.freeze
+  # SV wiki has three high ratings, all of which are rare:
+  # This is just a guess at appropriate weighting for the case where almost
+  # all articles are the lowest tier.
+  SVWIKI_WEIGHTING = { 'u' => 100,
+                       'b' => 90,
+                       'r' => 80,
+                       's' => 0 }.freeze
+  NLWIKI_WEIGHTING = { 'A' => 100,
+                       'B' => 75,
+                       'C' => 50,
+                       'D' => 25,
+                       'E' => 0 }.freeze
+  WEIGHTING_BY_LANGUAGE = {
+    'en' => ENWIKI_WEIGHTING,
+    'simple' => ENWIKI_WEIGHTING,
+    'fa' => ENWIKI_WEIGHTING,
+    'eu' => ENWIKI_WEIGHTING,
+    'fr' => FRWIKI_WEIGHTING,
+    'tr' => TRWIKI_WEIGHTING,
+    'ru' => RUWIKI_WEIGHTING,
+    'uk' => UKWIKI_WEIGHTING,
+    'gl' => ENWIKI_WEIGHTING,
+    'sv' => SVWIKI_WEIGHTING,
+    'nl' => NLWIKI_WEIGHTING,
+    'pt' => PTWIKI_WEIGHTING
+  }.freeze
+
+  def weighting
+    @weighting ||= WEIGHTING_BY_LANGUAGE[@wiki.language]
+  end
+
+  def weighted_mean_score(score)
```

Comment:
I like the idea! Added the new module on d0dfb96349ee564529ea94b2331a08fea71cd649, and modified the `LiftWingApi` class to include the module on 18c7a404fe4c14577c04b0d56edefeb448362703.
Didn't add specific specs for the new module because it's only used in the `LiftWingApi`, and the `LiftWingApi` 
 specs already check that the weighted mean score makes sense.